addons/website_sale_loyalty/static/src/js/checkout.js

```diff
@@ -19,7 +19,7 @@ websiteSaleCheckout.include({
             const cart_summary_discount_rewards = document.querySelectorAll(
                 '[data-reward-type=discount]'
             );
-            if (cart_summary_discount_rewards.length !== result.discount_reward_amounts.length) {
+            if (cart_summary_discount_rewards.length > result.discount_reward_amounts.length) {
```

Comment:
What happens if there is less discount line than expected?

Reply:
Not sure how that'd happen, but worst case scenario, the cart summary display is slightly off, perhaps missing one of the applied rewards as not all of the back-end's `discount_reward_amounts` values were used in this loop: https://github.com/odoo/odoo/blob/2ffc1d16dc0fa45ef2bc4b27e814c7ca241fe9af/addons/website_sale_loyalty/static/src/js/checkout.js#L26-L28

Reply:
I believe there's only two scenarios where this might happen:
1. allowing `promotion` type programs which trigger on a delivery product to add a discount[^1]
2. adding the controller's `_apply_reward` method to a route that doesn't trigger a refresh/redirect


[^1]: it's possible to create a program like this, but it doesn't do anything on either the web shop or the SO form

Reply:
I fear that it could lead to inconsistencies we didn't foresee.

If it's mandatory to fix the bug, or if you're confident it's the right thing to do, keep it.

Reply:
Changed it to instead simply `return` if no matches were found for `[data-reward-type=discount]`, which should also fix the issue for clients with outdated templates.

Could still be some risk of inconsistencies, but it should be way smaller I believe.